### DISCUSSION TITLE
sos1 - special ordered set type 1 constraint

### DISCUSSION THREAD

**Discussion by maolingao** [yalmip/YALMIP#1042] [Possible bug or missing functionality]
Hi Johan,

I would like to make use of the SOS1 constraint, however I only find the SOS2 supported yalmip: https://yalmip.github.io/command/sos2/

is there an easy way that I can employ the SOS1 of the underlying solver (f.e. gurobi) through yalmip?  thanks!

Model.addSOS() 
https://www.gurobi.com/documentation/9.5/refman/py_model_addsos.html#pythonmethod:Model.addSOS

Best,
Maolin

**Comment by Megumin715** [comment]
Hello, I have the same problem. How do you solve it

**Reply by maolingao** [reply]
I don't think SOS1 is available in yalmip yet. One possible way is try to use gurobi matlab api (https://www.gurobi.com/documentation/10.0/refman/matlab_the_model_argument.html), which is a bit complicated to implement for the first time (than yalmip).

**Reply by johanlofberg** [reply]
Correct, not supported so manual model using big-M required.
